pkg/pkgmiddleware/requestId.go

```diff
@@ -0,0 +1,65 @@
+// Copyright (C) 2022 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package pkgmiddleware
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/spacemonkeygo/monkit/v3"
+)
+
+// Key to use when setting the request ID.
+type ctxKeyRequestID string
+
+const (
+	// XStorjRequestID is the header key for the request ID.
+	XStorjRequestID = "X-Storj-Request-Id"
+
+	// RequestIDKey is the key that holds the unique request ID in a request context.
+	RequestIDKey ctxKeyRequestID = "request-id"
+)
+
+// AddRequestID uses XStorjRequestId field to set unique request Ids
+// in the response headers for each request of auth and linksharing service, if it dosen't alreasy exist.
+func AddRequestID(h http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		ctx := r.Context()
+		if w.Header().Get(XStorjRequestID) == "" {
```

Comment:
i think this should be `r.Header().Get(...` because we want to take the ID from an incoming request if it was given. I was thinking something like:

```
ctx := r.Context()
if r.Header().Get(XStorjRequestID) != "" {
  ctx = context.WithValue(ctx, RequestIDKey, r.Header().Get(XStorjRequestID))
  h.ServeHTTP(w, r.WithContext(ctx))
  return
}

trace := monkit.NewTrace(monkit.NewId())
requestID := fmt.Sprintf("%x", trace.Id())
w.Header().Set(XStorjRequestID, requestID)
ctx = context.WithValue(ctx, RequestIDKey, requestID)
h.ServeHTTP(w, r.WithContext(ctx))
```